File: statusupdate/distro.php
<?php

/* The distro listens on PGM, and republishes the messages to it's local clients */

$context = new ZMQContext();
$in = new ZMQSocket($context, ZMQ::SOCKET_SUB);
$in->setSockOpt(ZMQ::SOCKOPT_SUBSCRIBE, '');
$in->setSockOpt(ZMQ::SOCKOPT_RATE, 10000);
$in->connect("epgm://;239.192.0.1:7601");
$out = new ZMQSocket($context, ZMQ::SOCKET_PUB);
$out->bind("ipc:///tmp/events");

$device = new ZMQDevice(ZMQ::DEVICE_FORWARDER, $in, $out);

File: statusupdate/distro.c
#include <zmq.h>

int main(void) {
  void *ctx = zmq_init(1);
  void *in = zmq_socket(ctx, ZMQ_SUB);
  void *out = zmq_socket(ctx, ZMQ_PUB);
  zmq_setsockopt(in, ZMQ_SUBSCRIBE, "", 0);
  int rcin = zmq_connect(in, "epgm://;239.192.0.1:7601");
  int rcout = zmq_bind(out, "ipc:///tmp/events");
  
  int rcd = zmq_device(ZMQ_FORWARDER, in, out);
  
  zmq_close(in); zmq_close(out); zmq_term(ctx);
  return 0;
}